### ISSUE TITLE
Playing audio on iOS – "Seeking interrupted", short sound

### ISSUE THREAD

**Issue by MikePodgorniy** [expo/expo#223]
After first play, when trying to re-play the short sound, the error occurs again
everything works well on android
I use soundObject.stopAsync() or soundObject.playFromPositionAsync(0)
## On iOS
**durationMillis":0,"playableDurationMillis":0**
"volume":1,"shouldCorrectPitch":false,"uri":"https://www.englishdom.com/staticus/child.mp3","positionMillis":0,"isBuffering":false,"durationMillis":0,"playableDurationMillis":0,"progressUpdateIntervalMillis":500,"isMuted":false,"isLooping":false,"isPlaying":true,"shouldPlay":true,"rate":1,"isLoaded":true,"didJustFinish":false}

## on Android
**"playableDurationMillis":810,"durationMillis":810**
{"isLooping":false,"volume":1,"shouldCorrectPitch":false,"isLoaded":true,"uri":"/staticus/child.mp3","shouldPlay":false,"isMuted":false,"progressUpdateIntervalMillis":500,"durationMillis":810,"positionMillis":0,"isPlaying":false,"playableDurationMillis":810,"isBuffering":false,"didJustFinish":false,"rate":1}
### Error log
[exp] SuperSound_1 PlaySound {"volume":1,"shouldCorrectPitch":false,"uri":"https://www.englishdom.com/staticus/child.mp3","positionMillis":0,"isBuffering":false,"durationMillis":0,"playableDurationMillis":0,"progressUpdateIntervalMillis":500,"isMuted":false,"isLooping":false,"isPlaying":true,"shouldPlay":true,"rate":1,"isLoaded":true,"didJustFinish":true}
[exp] {"line":2027,"column":24,"sourceURL":"http://192.168.0.56:19001/main.bundle?platform=ios&dev=true&strict=false&minify=false&hot=false&assetPlugin=expo/tools/hashAssetFiles","framesToPop":1,"code":"E_AV_SEEKING","nativeStackIOS":["0   Exponent                            0x0000000100aaea14 ABI17_0_0RCTJSErrorFromCodeMessageAndNSError + 92","1   Exponent                            0x0000000100a5aa90 ABI17_0_0RCTParseMethodSignature + 11592","2   Exponent                            0x00000001009d7164 ReactABI17_0_0regionAsJSON + 137884","3   AVFoundation                        0x00000001890039b8 <redacted> + 160","4   AVFoundation                        0x00000001890139c8 <redacted> + 5400","5   libdispatch.dylib                   0x00000001804fe9e0 <redacted> + 24","6   libdispatch.dylib                   0x00000001804fe9a0 <redacted> + 16","7   libdispatch.dylib                   0x00000001805035e8 _dispatch_main_queue_callback_4CF + 996","8   CoreFoundation                      0x00000001815f50c8 <redacted> + 12","9   CoreFoundation                      0x00000001815f2ce4 <redacted> + 1572","10  CoreFoundation                      0x0000000181522da4 CFRunLoopRunSpecific + 424","11  GraphicsServices                    0x0000000182f8c074 GSEventRunModal + 100","12  UIKit                               0x00000001877dd058 UIApplicationMain + 208","13  Exponent                            0x0000000100134c44 Exponent + 543812","14  libdyld.dylib                       0x000000018053159c <redacted> + 4"],"domain":"ABI17_0_0RCTErrorDomain","userInfo":{"NSLocalizedDescription":"Seeking interrupted."}}
[exp] SuperSound_1 sendStatus AUDIO_ERROR
[exp] 1 {"volume":1,"shouldCorrectPitch":false,"uri":"https://www.englishdom.com/staticus/child.mp3","positionMillis":0,"isBuffering":false,"durationMillis":0,"playableDurationMillis":0,"progressUpdateIntervalMillis":500,"isMuted":false,"isLooping":false,"isPlaying":true,"shouldPlay":true,"rate":1,"isLoaded":true,"didJustFinish":false}
``````

**Comment by gkufera** [comment]
Hi @MikePodgorniy!

What a strange bug. Could you post a code sample in which you encounter this error so I can debug? I am not sure how to contextualize the `PlaybackStatus`es you posted.

Thanks!

Greg

**Comment by MikePodgorniy** [comment]
open on iOS Simulator or Expo on iPhone `exp start --dev --lan`
```javascript
import Expo from 'expo';
import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { Constants, Audio } from 'expo';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPlayAgain: false
    };
  }

  _handlePlaySoundAsync = async () => {
    await Audio.setIsEnabledAsync(true);
    const sound = new Audio.Sound();
    await sound.loadAsync({
      uri: 'https://www.englishdom.com/staticus/child.mp3'
    });
    await sound.playAsync();
    this.sound = sound;
    this.setState({
      showPlayAgain: true
    });
  };

  _handleCrash = async () => {
    console.log('(͡• ͜ʖ ͡•)');
    await this.sound.playFromPositionAsync(0);
    console.log('UPS');
  };

  render() {
    const { showPlayAgain } = this.state;
    return (
      <View style={styles.container}>
        <Button title="Play a sound!" onPress={this._handlePlaySoundAsync} />
        {showPlayAgain &&
          <Button title="Play a crash!" onPress={this._handleCrash} />}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: Constants.statusBarHeight,
    backgroundColor: '#ecf0f1'
  }
});

Expo.registerRootComponent(App);

```

**Comment by terribleben** [comment]
Hi there, @gkufera informs me that this may be an edge case with the metadata header frames on your particular mp3 file. Possibly due to the way the stream is being loaded, or possibly because it's an unusual flavor of mp3 file, or maybe both. Probably the former case since your Android device is able to read it without a problem.

We can try and make the mp3 loading code more robust on iOS. Leaving this issue open until @gkufera can come to a resolution.

**Comment by gkufera** [comment]
Hi @MikePodgorniy --

I just finished updating the AV playback API (for the upcoming SDK) to clarify that some files are not able to read a duration or playable duration, like in this case. This should prevent a crash, and make `positionMillis` give you an accurate reading for these types of files, but it seems that the underlying iOS structure we are using to play this file does not want to let us seek. I also noticed that the file referenced in #277 also has this problem. I believe that it is, as @terribleben said, a combination of the factors he mentioned.

We will make another pass at improving this API user experience when implementing stream support for AV playback, since iOS seems to effectively be treating your mp3 files as streams. For now, however, I would apply the following workaround in your particular case:

- Is possible, use mp3s that are not formatted in this way.
- Otherwise, use the upcoming SDK, which should prevent any surprises from happening with mp3s which have unreadable durations and playable durations, and if/when seeking is impossible, reload the sound to play from the beginning.

Thank you! Let me know if you have any further problems.

Greg
